tinygrad/utils.py

```diff
@@ -1,23 +1,41 @@
+from hashlib import md5
+import os
+
 import numpy as np
+import requests
+from filelock import FileLock
+
 
 def layer_init_uniform(m, h):
   ret = np.random.uniform(-1., 1., size=(m,h))/np.sqrt(m*h)
   return ret.astype(np.float32)
 
+
+mnist_filename = "mnist.npz"
+mnist_url = "https://storage.googleapis.com/tensorflow/tf-keras-datasets/" + mnist_filename
+mnist_md5_hash = "8a61469f7ea1b51cbae51d4f78837e45"
+mnist_fp = os.path.join("/tmp", mnist_md5_hash)
+
+mnist_lock = FileLock(mnist_fp + ".lock")
```

Comment:
This is just to avoid a race condition where one process is trying to read the file while another is still writing